modules/Material/ActionBar.qml

```diff
@@ -203,6 +204,43 @@ Item {
             overflowMenu.close();
     }
 
+    /*!
+       \internal
+
+       put mousearea under everything blow
+     */    
+    MouseArea { 
+        anchors.fill: parent
+
+        property var previousPosition
+		property var enable : Storage.target.clientSideDecorations 
+
+        onPressed: {
+            previousPosition = Qt.point(mouseX, mouseY)
+        }
+
+        onDoubleClicked: {
+			if( enable ){
+				if( Storage.target.visibility != Window.FullScreen ){
+					Storage.target.visibility = Window.FullScreen
+				}else{
+					Storage.target.visibility = Window.Windowed
+				}
+			}
+        }
+
+        onPositionChanged: {
+            if (pressedButtons == Qt.LeftButton 
+				&& Storage.target.visibility != Window.FullScreen 
+				&& enable ) {
+                var dx = mouseX - previousPosition.x
+                var dy = mouseY - previousPosition.y
+                Storage.target.x = Storage.target.x + dx 
+                Storage.target.y = Storage.target.y + dy
```

Comment:
On X11 moving the window should be done using the [_NET_WM_MOVERESIZE message of the EWMH](https://specifications.freedesktop.org/wm-spec/wm-spec-latest.html#idm140200472659776) (i.e. it should be done by the window manager and not by the app itself). In GDK there is the function `gdk_window_begin_move_drag()` which does exactly this but so far I haven't found any equivalent on Qt.
The same goes for wayland where `wl_shell_surface_move()` or `xdg_surface_move()` should be used. I am doubtful your approach would even work on wayland as wayland AFAIK does not allow you to access or set global window coordinates.